embedchain/embedchain.py

```diff
@@ -36,52 +39,81 @@ def __init__(self, config: BaseAppConfig):
         self.is_docs_site_instance = False
         self.online = False
 
-    def add(self, data_type, url, metadata=None, config: AddConfig = None):
+    def add(self, source, data_type=None, metadata=None, config: AddConfig = None):
         """
         Adds the data from the given URL to the vector db.
         Loads the data, chunks it, create embedding for each chunk
         and then stores the embedding to vector database.
 
-        :param data_type: The type of the data to add.
-        :param url: The URL where the data is located.
+        :param source: The data to embed, can be a URL, local file or raw content, depending on the data type.
+        :param data_type: Optional. Automatically detected, but can be forced with this argument.
+        The type of the data to add.
         :param metadata: Optional. Metadata associated with the data source.
         :param config: Optional. The `AddConfig` instance to use as configuration
         options.
+        :return: source_id, a md5-hash of the source, in hexadecimal representation.
         """
         if config is None:
             config = AddConfig()
 
+        try:
+            DataType(source)
+            logging.warning(
+                f"It looks like you used '{source}', a data_type, as the `source` argument. Since v0.0.35, Embedchain is able to detect the data_type automatically, so the arguments of the `add` method switched places."  # noqa #E501
```

Comment:
I'm adding this, however it might be better to educate people that they should be using `DataType.PDF_FILE` from `models`, because this is the safest way.